src/client.rs

```diff
@@ -195,21 +197,37 @@ async fn run_data_channel<T: Transport>(args: Arc<RunDataChannelArgs<T>>) -> Res
     let mut conn = do_data_channel_handshake(args.clone()).await?;
 
     // Forward
-    match read_data_cmd(&mut conn).await? {
+    match read_data_cmd(&mut conn.get_reliable_stream()).await? {
         DataChannelCmd::StartForwardTcp => {
-            run_data_channel_for_tcp::<T>(conn, &args.local_addr).await?;
+            match conn {
+                TransportStream::StrictlyReliable(reliable) => {
+                    run_data_channel_for_tcp::<T, T::ReliableStream>(reliable, &args.local_addr).await?;
+                }
+                TransportStream::PartiallyReliable(_, unreliable) => {
+                    run_data_channel_for_tcp::<T, T::UnreliableStream>(unreliable, &args.local_addr).await?;
+                }
+            }
+
         }
         DataChannelCmd::StartForwardUdp => {
-            run_data_channel_for_udp::<T>(conn, &args.local_addr).await?;
+            match conn {
+                TransportStream::StrictlyReliable(reliable) => {
+                    run_data_channel_for_udp::<T, T::ReliableStream>(reliable, &args.local_addr).await?;
+                }
+                TransportStream::PartiallyReliable(_, unreliable) => {
+                    run_data_channel_for_udp::<T, T::UnreliableStream>(unreliable, &args.local_addr).await?;
```

Comment:
This may also not work as expect. `run_data_channel_for_udp` call `write_all` to write packets. And `write_all` can call `poll_write` multiple times and there's no guarantee that a packet passed to `write_all` will not be fragmented and passed to `poll_write` at a whole, which means it's possible multiple quic datagrams are used to carry one `UdpTraffic`, and some of them are lost, causing the deserialization of `UdpTraffic` to fail.